src/architecture/merkledb.md

```diff
@@ -0,0 +1,348 @@
+# Exonum MerkleDB
+
+Exonum MerkleDB is a persistent storage implementation based on
+[RocksDB][rocks-db]. MerkleDB provides APIs to work with merkelized data
+structures.
+
+MerkleDB is an object database. Objects represent the highest abstraction level
+for the [data storage](storage.md). All objects fall into the following groups:
+
+- **blobs** which represent sequences of bytes
+- **root objects** that do not have parents. These objects have UTF-8
+  identifiers, for example, "block", "state". Root objects can contain blob
+  items inside them.
+
+1. [Root Objects Types](#root-object-types) lists the types of data
```

Comment:
Weird enough, my local copy of `mkdocs` glues this list with a previous one, resulting in quite incomprehensible formatting. I'd suggest to insert a paragraph before this list (something like "The rest of the article is structured as follows") to get rid of this bug (?) and make the formatting more appealing in general.

Reply:
done